libheif/heif_items.cc

```diff
@@ -216,6 +216,7 @@ struct heif_error heif_context_add_item(struct heif_context* ctx,
 
   if (result && out_item_id) {
     *out_item_id = result.value;
+    ctx->context->get_heif_file()->set_primary_item_id(*out_item_id);
```

Comment:
But that will overwrite any other primary item id that was set before whenever a new item is added, right?

Reply:
And only if `out_item_id` is not NULL, which is quite unexpected.